### PR TITLE
Fix basic mobile responsiveness issues (Part 1)

### PR DESCRIPTION
**Pull request by heypoom**
Fixes https://github.com/metabase/shoppy/issues/62

Improves very basic usability on mobile, such as the navigation bar, site switcher and columns. This is only the first pass and so only fixes basic glaring responsiveness issues - and does not consider the design itself, I'll pair with Alessio to improve the design aesthetics and the rest.

# Demo

### CODE REVIEW

File: src/components/layout/SidebarLinks.tsx [hunk 1]
```diff
@@ -60,7 +73,10 @@ const renderLink = (link: SidebarLink, child?: boolean) => {
       )}
       defaultOpened={link.defaultOpened}
     >
-      {link.children && link.children.map((link) => renderLink(link, true))}
+      {link.children &&
+        link.children.map((link) =>
+          renderLink(link, { child: true, onLinkClick }),
```
Comment (WiNloSt) [inline src/components/layout/SidebarLinks.tsx:78]: nit: I'd rename this to sound more like a boolean, `child` or `children` could be confused with a React node.

```suggestion
          renderLink(link, { isChild: true, onLinkClick }),
```

**Review by WiNloSt — APPROVED** [review]
Looks good to me 👍

File: src/components/layout/SidebarLinks.tsx (hunk 1 above)
Comment (heypoom) [reply src/components/layout/SidebarLinks.tsx]: Nice catch, fixed in aa9757619c6cdc04af1f7102e4437be59b9f5afd. Thanks!

**Comment by heypoom** [comment]
Note: DO NOT MERGE THIS PR YET. There is a bug with the site switching logic, causing infinite page refreshes. Will fix that first.

**Comment by heypoom** [comment]
PS. we'll merge this as-is first, and work on design improvements in the follow-up PR per [this Notion document](https://www.notion.so/metabase/Mobile-Design-Pairing-with-Alessio-11569354c90180559b6be88aaee6539b?pvs=4).

**Comment by heypoom** [comment]
Merging this first -- I'll address further review comments (if any) in the next PR 👍🏻

**Pull request merged**